### PR TITLE
Add C++ 3.x releases into download page

### PR DESCRIPTION
**Pull request by BewareMyPower**
The C++ client version has been no longer consistent with the Pulsar version since Pulsar 2.11.0. We need to provide the download URLs for packages of new 3.x releases.

image

### CODE REVIEW

File: site2/website-next/pulsar-cpp/releases.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[
+  "3.0.0"
```
Comment (tisonkun) [inline site2/website-next/pulsar-cpp/releases.json:2]: This data file can be a duplicate with `site2/website-next/data/release-cpp.js`. I suppose you resolve data from there (as SSOT) and test/filter whether it's `>= 3.0.0`.

File: site2/website-next/src/pages/download.js [hunk 2]
```diff
@@ -327,6 +344,16 @@ export default function page(props) {
...
+          <header className="postHeader mt-12">
+            <h1>
+              <Translate>Pulsar C++ Client</Translate>
+            </h1>
+            <div>
+              <p>The Link column contains C++ packages for different systems and architectures.</p>
+            </div>
+            <hr />
+          </header>
+          <ReleaseTable data={pulsarCppReleaseInfo}></ReleaseTable>
```
Comment (tisonkun) [inline site2/website-next/src/pages/download.js:356]: You may provide a preview of this section.

**Review by tisonkun — CHANGES_REQUESTED** [review]
Commented above.

File: site2/website-next/pulsar-cpp/releases.json (hunk 1 above)
Comment (BewareMyPower) [reply site2/website-next/pulsar-cpp/releases.json:2]: Good suggestion. I will fix it soon.

File: site2/website-next/src/pages/download.js (hunk 2 above)
Comment (BewareMyPower) [reply site2/website-next/src/pages/download.js:356]: Updated.

File: site2/website-next/pulsar-cpp/releases.json (hunk 1 above)
Comment (BewareMyPower) [reply site2/website-next/pulsar-cpp/releases.json]: Addressed.

**Review by merlimat — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
Good to go.

Potential follow-ups:

1. Common lib to provide release info (and resolved to current version);
2. Add download links to DEB and RPM installer;
3. Update links in the C++ client library doc page.

**Comment by BewareMyPower** [comment]
> Add download links to DEB and RPM installer;

It might be a little hard. We have multiple packages now (DEB, RPM, APK for both x64 and arm64), add there will be releases for Windows or macOS in future. It's better to provide a guide and encourage users to download the proper packages.

> Update links in the C++ client library doc page.

I'm working on it at the moment and will open a PR soon.

**Pull request merged**